### DISCUSSION TITLE
is "Amazon Time Sync Service" used by default in AWS?

### DISCUSSION THREAD

**Discussion by kristianpaynova** [bottlerocket-os/bottlerocket#2601] [Q&A] [answered]
Don't see this mentioned anywhere, so hopefully someone has the answer. If I deploy an EKS cluster with a nodegroup using the AWS-provided bottlerocket AMI, will that use be configured to use Amazon Time Sync Service by default or is that something that would need to be configured manually (if so how?). Thanks.

I noticed that the ip 169.254.169.123 (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-time.html) is mentioned in [one of the config files](https://github.com/bottlerocket-os/bottlerocket/blob/e44527ce646e12dcdf31e6a158cbc1785857000a/sources/models/shared-defaults/defaults.toml#L109), but [in another it's excluded](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/shared-defaults/public-ntp.toml), and i'm not familiar enough with this code to know which defaults are used when.

**Comment by kristianpaynova** [comment, marked as answer]
Probably answering my own question here (but would be great if someone confirm)- seems like it _does_ get used by default as public-ntp.toml is not referenced in the aws config while defaults.toml is.

https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/models/src/aws-k8s-1.24/defaults.d

**Reply by stmcginnis** [reply]
Yep, you figured it out @kristianpaynova!

Those settings are associated with different variants by creating a symlink between the source toml file and the path under `sources/models/src/$VARIANT`. The defaults.toml file gets pulled in for each of the AWS variants, so when running with EKS or one of the other AWS-based variants they are automatically configured to use Amazon Time Sync.

When running elsewhere, the other variants pull in the `public-ntp.toml` settings you found:

```txt
find ./sources/models -name "*ntp.toml"
./sources/models/shared-defaults/public-ntp.toml
./sources/models/src/metal-dev/defaults.d/70-public-ntp.toml
./sources/models/src/vmware-dev/defaults.d/70-public-ntp.toml
./sources/models/src/metal-k8s-1.22/defaults.d/70-public-ntp.toml
./sources/models/src/metal-k8s-1.24/defaults.d/70-public-ntp.toml
./sources/models/src/vmware-k8s-1.24/defaults.d/70-public-ntp.toml
./sources/models/src/vmware-k8s-1.22/defaults.d/70-public-ntp.toml
```
